extensions/authy/src/component/OtpList.tsx

```diff
@@ -0,0 +1,148 @@
+    const [serviceTimer, setServiceTimer] = useState<number>(30);
+    const [appTimer, setAppTimer] = useState<number>(10);
+
+    async function refresh(): Promise<void> {
+        const toast = await showToast(ToastStyle.Animated, "Authy", "Refreshing");
+        await toast.show();
+        setState({apps: [], services: []});
+        try {
+            const {authyId} = getPreferenceValues<{ authyId: number }>();
+            const deviceId: number = await getFromCache(DEVICE_ID);
+            const secretSeed: string = await getFromCache(SECRET_SEED);
+            // get authy apps
+            const authyApp = await getAuthyApps(authyId, deviceId, genTOTP(secretSeed));
+            await addToCache(APPS_KEY, authyApp)
+            // get 3rd party services
+            const services = await getServices(authyId, deviceId, genTOTP(secretSeed));
+            await addToCache(SERVICES_KEY, services);
+        } catch (error) {
+            if (error instanceof Error) {
+                await showToast(ToastStyle.Failure, "Authy", error.message);
+                await popToRoot()
+            } else {
+                throw error;
+            }
+        }
+        await loadData();
+        await toast.hide();
+        await showToast(ToastStyle.Success, "Authy", "Data was successful refreshed");
+    }
+
+    async function loadData(): Promise<void> {
+        try {
+            if (!props.isLogin) {
+                return;
+            }
+            const servicesResponse: Services = await getFromCache(SERVICES_KEY);
+            const appsResponse: AuthyApp = await getFromCache(APPS_KEY);
+            const {authyPassword} = getPreferenceValues<{ authyPassword: string }>();
+            const services = servicesResponse.authenticator_tokens.map(i => {
+                const seed = decryptSeed(i.encrypted_seed, i.salt, authyPassword);
+                return {
+                    name: i.original_name ? i.original_name : i.name,
+                    logo: i.logo,
+                    digits: i.digits,
+                    seed: seed,
+                    otp: generateTOTP(seed, {digits: i.digits, period: 30})
+                }
+            });
+            const apps = appsResponse.apps.map(i => {
+                return {
+                    name: i.name,
+                    logo: "",
+                    digits: i.digits,
+                    seed: i.secret_seed,
+                    otp: generateTOTP(encode(Buffer.from(i.secret_seed, "hex")), {digits: i.digits, period: 10})
+                }
+            });
+            console.log(apps, services)
+            setState({apps: apps, services: services})
+        } catch (error) {
+            if (error instanceof Error) {
+                await showToast(ToastStyle.Failure, "Authy", error.message);
+            } else {
+                throw error;
+            }
+        }
+    }
+
+    useEffect(() => {
+        loadData()
+    }, [props.isLogin])
+
+    useInterval(() => {
```

Comment:
Not sure that this is a good idea, but my goal was to show how long OTP will be valid. This leads to 2 re-renders each second.